### PR TITLE
Chore: Refactor useModels to use react-query rather than useReducer

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Refactor `useModels`:

- rename it to `useContentTypes`
- replace useReducer with react-query (we were reinventing the wheel there)
- add tests

### Why is it needed?

In a next step I'd like to pass in query params, to fetch content-types sorted.

### CODE REVIEW

File: packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
+import { waitFor } from '@testing-library/react';
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js:4]: I don't think you should mix these, `waitFor` is returned from the `render` function of `@testing-library/react-hooks` – wdyt?

File: packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          },
+        },
+      });
+
+      return (
+        <QueryClientProvider client={client}>
+          <IntlProvider locale="en" messages={{}}>
+            {children}
+          </IntlProvider>
+          ,
```
[the remark this answers is not in the document]
Comment (gu-stav) [reply packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js:98]: Fixed & rebased

File: packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js (hunk 1 above)
Comment (gu-stav) [reply packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js]: Fixed and rebased. Good catch - didn't know the render function returns `waitFor`.

Comment (joshuaellis) [reply packages/core/admin/admin/src/hooks/useContentTypes/tests/useContentTypes.test.js]: only in `react-hooks` lol

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**